analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/SymbolicValue.cs

```diff
@@ -27,26 +27,16 @@ namespace SonarAnalyzer.SymbolicExecution.Roslyn
 {
     public sealed record SymbolicValue
     {
-        private readonly SymbolicValueCounter counter;
-        private readonly int identifier;    // This is debug information that is intentionally excluded from GetHashCode and Equals
         // SymbolicValue can have only one constraint instance of specific type at a time
         private ImmutableDictionary<Type, SymbolicConstraint> Constraints { get; init; } = ImmutableDictionary<Type, SymbolicConstraint>.Empty;
 
-        public SymbolicValue(SymbolicValueCounter counter)
-        {
-            this.counter = counter;
-            identifier = counter.NextIdentifier();
-        }
+        public SymbolicValue() { }
 
-        protected SymbolicValue(SymbolicValue original) // Custom record copying constructor
-        {
-            counter = original.counter;
-            identifier = counter.NextIdentifier();
+        protected SymbolicValue(SymbolicValue original) => // Custom record copying constructor
```

Comment:
I think the rule is right. Why have a `protected` ctor in a sealed class?

Reply:
Yes, you're right. It had to be there before because of the additional logic (and had to be `protected` because of how `record` code generation works). It doesn't need to be there anymore.